pennylane/ops/op_math/change_op_basis.py

```diff
@@ -177,9 +184,9 @@ def decomposition(self):
         r"""Decomposition of the product operator is given by each of compute_op, target_op, compute_op† applied in succession."""
         if queuing.QueuingManager.recording():
             return [
-                self[2]._unflatten(*self[2]._flatten()),  # pylint: disable=protected-access
-                self[1]._unflatten(*self[1]._flatten()),  # pylint: disable=protected-access
-                self[0]._unflatten(*self[0]._flatten()),  # pylint: disable=protected-access
+                pytrees.unflatten(*pytrees.flatten(self[2])),  # pylint: disable=protected-access
+                pytrees.unflatten(*pytrees.flatten(self[1])),  # pylint: disable=protected-access
+                pytrees.unflatten(*pytrees.flatten(self[0])),  # pylint: disable=protected-access
             ]
         return list(self[::-1])
```

Comment:
So `decomposition` has never been capture compatible, so I actually would say we don't need this here.  I would instead just do:
```
if QueuingManager.recording():
    _ = [op.queue() for op in reversed(self)]
```